test/unit/test_disco_config.py

```diff
@@ -100,3 +100,121 @@ def test__path_exists__passthrough_successful(self, open_mock):
         found = disco_config.open_normalized("path", "to", "file", mode="moody")
         self.assertIs(expected, found)
         open_mock.assert_called_once_with("FAKE_CONFIG_DIR/path/to/file", mode="moody")
+
+
+class TestAsiaqConfig(TestCase):
+    """Tests for the AsiaqConfig object."""
+    # allow long method names
+    # pylint: disable=invalid-name
+    BASE_CONFIG_DICT = {
+        disco_config.DEFAULT_CONFIG_SECTION: {
+            'default_environment': 'fake-build',
+            'default_unused_option': 'fall-all-the-way-back',
+        },
+        'mhcfoobar': {
+            'easy_option': 'easy_answer',
+            'envy_option': 'fallback_answer',
+            'envy_option@fake-build': 'default_env_answer',
+            'envy_option@ci': 'ci_answer'
+        }
+    }
+
+    S3_BUCKET_CONFIG = {
+        's3_bucket_base': 'bucket-base',
+        's3_bucket_suffix': 'blah',
+        's3_bucket_suffix@production': 'danger'
+    }
+
+    def test__get_asiaq_option__no_env_options(self):
+        "Option exists in desired section: found it"
+        config = MockAsiaqConfig(deepcopy(self.BASE_CONFIG_DICT))
+        self.assertEquals('easy_answer', config.get_asiaq_option(option='easy_option', section='mhcfoobar'))
+
+    def test__get_asiaq_option__default_env(self):
+        "Env-specific option with default environment"
+        config = MockAsiaqConfig(deepcopy(self.BASE_CONFIG_DICT))
+        self.assertEquals('default_env_answer',
+                          config.get_asiaq_option(option='envy_option', section='mhcfoobar'))
+
+    def test__get_asiaq_option__env_in_constructor(self):
+        "Env-specific option with environment passed in at construction time"
+        config = MockAsiaqConfig(deepcopy(self.BASE_CONFIG_DICT), environment='ci')
+        self.assertEquals('ci_answer',
+                          config.get_asiaq_option('envy_option', section='mhcfoobar'))
+
+    def test__get_asiaq_option__env_in_call(self):
+        "Env-specific option with environment passed in at call time"
+        config = MockAsiaqConfig(deepcopy(self.BASE_CONFIG_DICT))
+        self.assertEquals('ci_answer',
+                          config.get_asiaq_option('envy_option', section='mhcfoobar', environment='ci'))
```

Comment:
Nit: I may have missed it, but I didn't notice a test testing the behavior when environment is set at construction time and also passed in at call time.